src/goodbyedpi.c

```diff
@@ -98,13 +94,13 @@ WINSOCK_API_LINKAGE INT WSAAPI inet_pton(INT Family, LPCSTR pStringBuf, PVOID pA
 static int running_from_service = 0;
 static HANDLE filters[MAX_FILTERS];
 static int filter_num = 0;
-static const char *http10_redirect_302 = "HTTP/1.0 302 ";
-static const char *http11_redirect_302 = "HTTP/1.1 302 ";
-static const char *http_host_find = "\r\nHost: ";
-static const char *http_host_replace = "\r\nhoSt: ";
-static const char *http_useragent_find = "\r\nUser-Agent: ";
-static const char *location_http = "\r\nLocation: http://";
-static const char *connection_close = "\r\nConnection: close";
+static const char http10_redirect_302[] = "HTTP/1.0 302 ";
+static const char http11_redirect_302[] = "HTTP/1.1 302 ";
+static const char http_host_find[] = "\r\nHost: ";
+static const char http_host_replace[] = "\r\nhoSt: ";
+static const char http_useragent_find[] = "\r\nUser-Agent: ";
+static const char location_http[] = "\r\nLocation: http://";
+static const char connection_close[] = "\r\nConnection: close";
```

Comment:
>This will copy the data once more.

I don't understand what you mean. Where do you suppose it would copy the data one more time?

Reply:
In the memory. `[]` would put a string in `.rdata` section and copy it to `.data` section in runtime, you are allowed to modify data inside this variable. While `*` will keep this variable in `.rdata` section which is read-only, won't copy it in runtime, and you're not allowed to modify it (since it's read-only and you'll get memory violation, not because of `const`).
It's better to define static variables with `strlen()` of these variables if you feel like it.

Reply:
1 the strings are declared as const char[].
2 The compiler actually puts them into rdata.